### PR TITLE
Backlog: Copy XSA Key Files

### LINKED ISSUE #385: Backlog: Dynamic look for DAT/KEY file
**Problem statement**
Currently the path to DAT/KEY files are hard-coded.

**Enhancement**
1. Set abs path for DAT/KEY file:
- hsr_dat_name
- hsr_key_name

**Notes**

1. See also related #426 

**Dependencies**

**Checklist**

### LINKED ISSUE #426: XSA DAT/KEY files need to be copied across cluster nodes
**Problem statement**

As originally noted in the review of #373 and subsequently on #385, the HANA cluster copying of the DAT/KEY file for xs are currently missing.

**Enhancement**
1. Copy DAT/KEY files for xs across cluster nodes:
- `/usr/sap/<SID>/SYS/global/xsa/security/ssfs/data/SSFS_<SID>.DAT`
- `/usr/sap/<SID>/SYS/global/xsa/security/ssfs/key/SSFS_<SID>.KEY`

**Notes**

**Dependencies**

**Checklist**

### PR DESCRIPTION
**Pull request by robfallows**
## Problem

Closes #426

Closes #385

## Description

This PR:

- Refactors the `copy_ssfs_keys.yml` into two tasks to simplify and extend usability.
- Adds a new task, `set_runtime_path_facts.yml`, to determine the SAP HANA global path at runtime.
- Removes the role `vars` which are now computed at runtime.

## Pre-Requisites

- A HANA cluster with HSR and XSA.
- Log in to the RTI.

## Commitment to Test

- Testing takes 5 minutes.
- Reviewing changed files.

## Test Instructions / Acceptance Criteria

- [ ] Code changes are acceptable.

Using the RTI, confirm the following:

- [ ] Check HSR: Running `ansible -i ~/hosts 10.1.1.4 -m shell -a "sudo su - hn1adm -c 'hdbnsutil -sr_state'  2>/dev/null | grep '^Operation'" --become 2>/dev/null` reports:

  ```text
  10.1.1.4 | CHANGED | rc=0 >>
  Operation mode of SITEA: primary
  Operation mode of SITEB: logreplay
  ```

For the following checks, ensure the SHA1 keys are the same for each host. Examples are shown, below.

- [ ] Check main SSFS.DAT files are the same: Running `ansible -i ~/hosts hanadbnodes -m shell -a "sha1sum /usr/sap/HN1/SYS/global/security/rsecssfs/data/SSFS_HN1.DAT" --become 2>/dev/null | grep SSFS` reports:

  ```text
  e3f42924a072e59d0688b89d437443f0a7fe56c7  /usr/sap/HN1/SYS/global/security/rsecssfs/data/SSFS_HN1.DAT
  e3f42924a072e59d0688b89d437443f0a7fe56c7  /usr/sap/HN1/SYS/global/security/rsecssfs/data/SSFS_HN1.DAT
  ```

- [ ] Check main SSFS.KEY files are the same: Running `ansible -i ~/hosts hanadbnodes -m shell -a "sha1sum /usr/sap/HN1/SYS/global/security/rsecssfs/key/SSFS_HN1.KEY" --become 2>/dev/null | grep SSFS` reports:

  ```text
  2d7b314445784cfb19e009d3db6ffac5e30dd112  /usr/sap/HN1/SYS/global/security/rsecssfs/key/SSFS_HN1.KEY
  2d7b314445784cfb19e009d3db6ffac5e30dd112  /usr/sap/HN1/SYS/global/security/rsecssfs/key/SSFS_HN1.KEY
  ```
- [ ] Check XSA SSFS.DAT files are the same: Running `ansible -i ~/hosts hanadbnodes -m shell -a "sha1sum /usr/sap/HN1/SYS/global/xsa/security/ssfs/data/SSFS_HN1.DAT" --become 2>/dev/null | grep SSFS` reports:

  ```text
  bc3f37f75165e98697ae7e558e8136b17d57ba96  /usr/sap/HN1/SYS/global/xsa/security/ssfs/data/SSFS_HN1.DAT
  bc3f37f75165e98697ae7e558e8136b17d57ba96  /usr/sap/HN1/SYS/global/xsa/security/ssfs/data/SSFS_HN1.DAT
  ```

- [ ] Check XSA SSFS.KEY files are the same: Running `ansible -i ~/hosts hanadbnodes -m shell -a "sha1sum /usr/sap/HN1/SYS/global/xsa/security/ssfs/key/SSFS_HN1.KEY" --become 2>/dev/null | grep SSFS` reports:

  ```text
  2d7b314445784cfb19e009d3db6ffac5e30dd112  /usr/sap/HN1/SYS/global/security/rsecssfs/key/SSFS_HN1.KEY
  2d7b314445784cfb19e009d3db6ffac5e30dd112  /usr/sap/HN1/SYS/global/security/rsecssfs/key/SSFS_HN1.KEY
  ```

### CODE REVIEW

File: deploy/v2/ansible/roles/hana-system-replication/tasks/copy_ssfs_keys.yml [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
 
+- name: Ensure the Primary node SSFS_{{ sid | upper }}.DAT file exists
+  when: ansible_hostname == hana_database.nodes[0].dbname
+  stat:
+    path: "{{ path_ssfs_dat }}"
+  register: primary_dat_file_result
+  failed_when: primary_dat_file_result.failed
+
+- name: Ensure the Primary node SSFS_{{ sid | upper }}.KEY file exists
+  stat:
```
[the remark this answers is not in the document]
Comment (robfallows) [reply deploy/v2/ansible/roles/hana-system-replication/tasks/copy_ssfs_keys.yml]: Good catch. Fixed.

File: deploy/v2/ansible/roles/hana-system-replication/tasks/copy_ssfs_key_path.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+---
```
Comment (rukawata) [inline deploy/v2/ansible/roles/hana-system-replication/tasks/copy_ssfs_key_path.yml:1]: The name of this task `copy_ssfs_key_path` is a bit confusing, especially when someone later read `copy_ssfs_keys`.
Maybe `copy_per_ssfs_key`, or `copy_ssfs_key_per_path`, or `copy_single_ssfs_key`?

Comment (robfallows) [reply deploy/v2/ansible/roles/hana-system-replication/tasks/copy_ssfs_key_path.yml]: Code updated to use `copy_single_ssfs_key`: a good name!

File: deploy/v2/ansible/roles/hana-system-replication/tasks/copy_ssfs_key_path.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    # it appears that the RTI attempts to resolve the hostname itself and fails.
+
+    - name: Ensure the primary node {{ ssfs_file_path }} file is placed on the secondary node
+      synchronize:
+        set_remote_user: false
+        src: "{{ ssfs_file_path }}"
+        dest: "{{ ssfs_file_path }}"
+        mode: push
+        archive: true
+      delegate_to: "{{ hana_database.nodes[0].ip_admin_nic }}"
```
[the remark this answers is not in the document]
Comment (robfallows) [reply deploy/v2/ansible/roles/hana-system-replication/tasks/copy_ssfs_key_path.yml]: That's in a lot of places in the code, not just in this role. Is there any chance that could go on the backlog?

**Comment by rukawata** [comment]
I have to revert this PR since it breaks pipeline (issue #466).
Please create a new PR with branch `centiq-backlog-xsa-key-files` and address the issue.

Thanks!

**Pull request merged**